### PR TITLE
[Remove Stream/Optional usage] LPS-172269 Remove usages of Stream/Optional on Method Return Value for module: portal-odata-impl

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:headless

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:headless

**Comment by W2jx** [comment]
rebase and resolve conflicts

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java [hunk 1]
```diff
@@ -351,11 +352,8 @@ else if (Objects.equals(BinaryExpression.Operation.NE, operation)) {
 		else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 			filter = _getORFilter((Filter)left, (Filter)right);
 		}
-		else {
-			return Optional.empty();
-		}
 
-		return Optional.of(filter);
+		return filter;
 	}
```
Comment (yuhai) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java:358]: @W2jx, please add SF to inline the method _getFilter()

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ExpressionVisitorImpl.java [hunk 2]
```diff
@@ -298,35 +297,35 @@ else if (uriResource instanceof UriResourcePartTyped) {
 				uriResources);
 	}
 
-	private Optional<BinaryExpression.Operation> _getOperationOptional(
+	private BinaryExpression.Operation _getOperation(
 		BinaryOperatorKind binaryOperatorKind) {
```
Comment (yuhai) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ExpressionVisitorImpl.java:301]: @W2jx please add SF to inline the method _getOperation()

File: modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/ExpressionConvertImplTest.java [hunk 3]
```diff
@@ -281,16 +279,20 @@ public void testConvertMemberExpressionWithLambdaAnyEqOnCollectionField()
...
+				"dateTime", locale -> "dateTime", locale -> "dateTime");
+
+			return HashMapBuilder.put(
+				collectionEntityField.getName(), collectionEntityField
+			).put(
+				stringEntityField.getName(), stringEntityField
+			).put(
+				dateTimeEntityField.getName(), dateTimeEntityField
+			).build();
 		}
```
Comment (yuhai) [inline modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/ExpressionConvertImplTest.java:297]: @W2jx, for this commit and the other similar commits, please inline these. We can know `collectionEntityField.getName()` value.

**Comment by yuhai** [comment]
@W2jx, the codes looks good to me. Please check my above comments and just do some SF.

**Comment by W2jx** [comment]
Hi @yuhai , I have modified the code as per the comments.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:headless

**Comment by W2jx** [comment]
no conflicts(1/26 10:20)

**Review by yuhai — APPROVED** [review]
@dantewang, the codes look good to me.

**Comment by W2jx** [comment]
no conflicts(1/30 13:06)

**Comment by dantewang** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/129382

**Pull request closed without merging**